pkg/secrets-store/nodeserver.go

```diff
@@ -203,6 +205,18 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 
 		log.Infof("Calling provider: %s", providerName)
 		providerBinary := fmt.Sprintf("%s/%s/provider-%s", providerVolumePath, providerName, providerName)
+
+		// check if provider is compatible with driver
+		if v, exists := ns.minProviderVersions[providerName]; exists {
```

Comment:
return an error when providerName does not exist?

Reply:
I was thinking if the compatible version isn't explicitly defined then we can assume all provider versions are compatible with the driver. But after a little more thought, I think we should explicitly define the versions in the deployment and error out here to avoid scenarios where deployment manifest is directly updated with the new image instead of applying the whole new manifest.

Reply:
I mean I did not see any handling of if  the providername does not exist in the map.

Reply:
@ritazh I haven't pushed all the changes yet. Once I push that change CI will fail until we update the helm chart with the new min provider version flag.

Reply:
done!